runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -292,7 +293,7 @@ impl Wasmer2VM {
         artifact: &VMArtifact,
         mut import: Wasmer2Imports<'_, '_, '_>,
         method_name: &str,
-    ) -> Result<(), VMError> {
+    ) -> Result<(), Result<FunctionCallError, VMRunnerError>> {
```

Comment:
Wrong nesting: 

```
Result<Result<(), FunctionCallError>, VMRunnerError>
```

Also, aling this with precompile's signature (in a follow up perhaps): both should either use Option or Result<(). My vote is for the latter.

Reply:
Conceptually, your nesting makes more sense. But in terms of yeeting `?`, it seems to be worse if I am not missing something....

The more common case is that we want to early return because of a `FunctionCallError`. This works with a `map_err(..)?` in the old (wrong) nesting. In the correct nesting, this is awkward because we want to early return on `Ok(Err(fn_call_error))` which is not something that `?` facilitates.

This translates the `?` I have now to something like this: 
```rust
// when foo() returns `Result<(), FunctionCallError>,`
let inner = match foo() {
    Ok(inner) => inner,
    Err(user_error => {
        return Ok(Err(user_error));
    }
};
```
Still trying to find a nicer solution here...

Reply:
Hm, yeah... I think that `Result<Result<(), FunctionCallError>, VMRunnerError>` is the conceptually right signature, and that we should use that troughout. Maybe even via `VMRunnerResult<T>` alias.

And yeah, this does require writing explicit matches for the inner result, but I think that's an Ok price to pay here (pun not intended). The inner thing is a value, it should be handled as value. Only the outer thing is semantically correct to through.